qiskit_nature/operators/second_quantization/fermionic_op.py

```diff
@@ -397,9 +404,11 @@ def to_normal_order(self) -> "FermionicOp":
                     True,
                 )
 
+        self.sparse_label = temp_sparse_label
+
         if isinstance(ret, FermionicOp):
             return ret
-        return FermionicOp(("", 0), self.register_length, True)
+        return FermionicOp(("", 0), self.register_length, self.sparse_label)
```

Comment:
Is this assuming that the current FermionicOp is sparse. I assumed the setting its sparse labal to False we because logic was used that depended on that. Hence whether it was False or not that set it as wanted. The label is later restored as it was originally. Now the method says it returns a sparse op, yet this line was changed from setting it True to be whatever is in the current operator. Might that be False? Or is this method assuming its sparse - in which case is that ok if its called on an operator that is not sparse - or should that raise an error.

Reply:
Current FermionicOp is dense in default. I personally had a hard time with the implementation here, and I thought it would be more convenient to return the same type as the original operator, so I did this. But the doc string of the method said "it returns sparse label", so this should be True as you said. Thanks.